src/core_init_atmosphere/mpas_geotile_manager.F

```diff
@@ -0,0 +1,1025 @@
+    !
+    !-----------------------------------------------------------------------
+    subroutine mpas_geotile_mgr_tile_to_latlon(mgr, tile, j, i, lat, lon)
+
+        implicit none
+
+        class (mpas_geotile_mgr_type) :: mgr
+        type (mpas_geotile_type), pointer, intent(in) :: tile
+        integer, value :: j
+        integer, value :: i
+        real (kind=RKIND), intent(out) :: lat
+        real (kind=RKIND), intent(out) :: lon
+
+        integer, pointer :: tile_bdr
+        real (kind=RKIND), pointer :: known_lon
+        real (kind=RKIND), pointer :: known_lat
+        real (kind=RKIND), pointer :: dx
+        real (kind=RKIND), pointer :: dy
+        integer :: ierr
+
+        ierr = 0
+
+        call mpas_pool_get_config(mgr % pool, 'tile_bdr', tile_bdr)
+        call mpas_pool_get_config(mgr % pool, 'known_lat', known_lat)
+        call mpas_pool_get_config(mgr % pool, 'known_lon', known_lon)
+        call mpas_pool_get_config(mgr % pool, 'dx', dx)
+        call mpas_pool_get_config(mgr % pool, 'dy', dy)
+
+        lat = real((j - (tile_bdr + 1) + tile % y - 1), kind=RKIND) * dy + known_lat + 1.0
+        lon = real((i - (tile_bdr + 1) + tile % x - 1), kind=RKIND) * dx + known_lon + 1.0
+
+        call deg2Rad(lat)
+        call deg2Rad(lon)
+
+    end subroutine mpas_geotile_mgr_tile_to_latlon
+
+
+    !***********************************************************************
+    !
+    !  public subroutine mpas_geotile_mgr_latlon_to_pixel => latlon_to_pixel
+    !
+    !> \brief   Translate a latitude, longitude coordinate to pixel coordinates
+    !> \author  Miles A. Curry
+    !> \date    02/20/2020
+    !> \details
+    !> Translate a latitude, longitude coordinate into global pixel coordinates. lat
+    !> should be in the range of -.5 * pi to .5 * pi and lon should be between -pi
+    !> and pi.
+    !
+    !-----------------------------------------------------------------------
+    subroutine mpas_geotile_mgr_latlon_to_pixel(mgr, lat, lon, pixel_x, pixel_y)
+
+        implicit none
+
+        class (mpas_geotile_mgr_type) :: mgr
+        real (kind=RKIND), value :: lat
+        real (kind=RKIND), value :: lon
+        integer, intent(out) :: pixel_x
+        integer, intent(out) :: pixel_y
+
+        integer, pointer :: tile_bdr
+        real (kind=RKIND), pointer :: known_lon
+        real (kind=RKIND), pointer :: known_lat
+        real, pointer :: dx
+        real, pointer :: dy
+        integer :: ierr
+
+        ierr = 0
+
+        call mpas_pool_get_config(mgr % pool, 'tile_bdr', tile_bdr)
+        call mpas_pool_get_config(mgr % pool, 'known_lon', known_lon)
+        call mpas_pool_get_config(mgr % pool, 'known_lat', known_lat)
+        call mpas_pool_get_config(mgr % pool, 'dx', dx)
+        call mpas_pool_get_config(mgr % pool, 'dy', dy)
+
+        call rad2Deg(lat)
+        call rad2Deg(lon)
+
+        pixel_x = int((lon - known_lon) / dx)
+        pixel_y = int((lat - known_lat) / dy)
```

Comment:
In lines 692 and 693, I think `int` should be `nint`.

Reply:
Trying out `nint` (as well as `floor` and `ceiling`) will cause the geotile_manager to generate tile names that don't exist for the dataset. Although, I'm not sure if this behavior is correct or not. For instance the geotile_manager will generate the tile `42001-43200.21601-22800` for the point: ```(1.57080, 0.00000)```. The calculation is:

```
* lat = 90.0
* (lat - known_lat) / dx = 21599.5059
* nint(21599.5059) = 21600
```

However, I don't think this is correct, as the very last tile contains the `21600` pixel: `42001-43200.20401-21600`. 

So perhaps the tile filename generation is incorrect? Perhaps off by one?

Reply:
Are the pixel coordinates that are returned by `mpas_geotile_mgr_latlon_to_pixel` supposed to be 0-based or 1-based?

Reply:
Let's assume that every latitude value in the interval [-Pi/2, Pi/2] is uniquely mapped to a j-index. Then for the center latitude, *lat*, of a given pixel, we can either say that all latitudes in the interval [*lat* - Pi/2, *lat* + Pi/2) map to that pixel, or that all latitudes in the interval (*lat* - Pi/2, *lat* + Pi/2] map to that pixel. Regardless of which we choose, there will always be a pole, i.e., -Pi/2 or Pi/2 that is not mapped to some j-index and that therefore needs special treatment.